crates/chainio/clients/avsregistry/src/writer.rs

```diff
@@ -194,6 +195,153 @@ impl AvsRegistryChainWriter {
+        quorum_numbers: Bytes,
+        socket: String,
+        operators_to_kick: Vec<Address>,
+        churn_signer_private_key: String,
+        churn_sig_salt: FixedBytes<32>,
+        churn_sig_expiry: U256,
+    ) -> Result<TxHash, AvsRegistryError> {
+        let provider = get_signer(&self.signer.clone(), &self.provider);
+        let operator_wallet = PrivateKeySigner::from_str(&self.signer)
+            .map_err(|_| AvsRegistryError::InvalidPrivateKey)?;
+        let operator_address = operator_wallet.address();
+
+        info!(
+            avs_service_manager = %self.service_manager_addr,
+            operator = %operator_address,
+            quorum_numbers = ?quorum_numbers,
+            "registering operator with churn the AVS's registry coordinator"
+        );
+
+        let contract_registry_coordinator =
+            RegistryCoordinator::new(self.registry_coordinator_addr, &provider);
+
+        let g1_hashed_msg_to_sign = contract_registry_coordinator
+            .pubkeyRegistrationMessageHash(operator_address)
+            .call()
+            .await
+            .map_err(|_| AvsRegistryError::PubKeyRegistrationMessageHash)?
+            ._0;
+
+        let sig = bls_key_pair
+            .sign_hashed_to_curve_message(alloy_g1_point_to_g1_affine(g1_hashed_msg_to_sign))
+            .g1_point();
+        let alloy_g1_point_signed_msg = convert_to_g1_point(sig.g1())?;
+        let g1_pub_key_bn254 = convert_to_g1_point(bls_key_pair.public_key().g1())?;
+        let g2_pub_key_bn254 = convert_to_g2_point(bls_key_pair.public_key_g2().g2())?;
+
+        let pub_key_reg_params = PubkeyRegistrationParams {
+            pubkeyRegistrationSignature: alloy_g1_point_signed_msg.clone(),
+            pubkeyG1: g1_pub_key_bn254.clone(),
+            pubkeyG2: g2_pub_key_bn254.clone(),
+        };
+
+        let msg_to_sign = self
+            .el_reader
+            .calculate_operator_avs_registration_digest_hash(
+                operator_address,
+                self.service_manager_addr,
+                operator_to_avs_registration_sig_salt,
+                operator_to_avs_registration_sig_expiry,
+            )
+            .await?;
+
+        let operator_signature = operator_wallet
+            .sign_hash(&msg_to_sign)
+            .await
+            .map_err(|_| AvsRegistryError::InvalidSignature)?;
+
+        let operator_signature_with_salt_and_expiry = SignatureWithSaltAndExpiry {
+            signature: operator_signature.as_bytes().into(),
+            salt: operator_to_avs_registration_sig_salt,
+            expiry: operator_to_avs_registration_sig_expiry,
+        };
+
+        let operators_to_kick_params: Vec<OperatorKickParam> = operators_to_kick
+            .iter()
+            .map(|address| OperatorKickParam {
+                operator: *address,
+                quorumNumber: 0_u8, // TODO: manejar múltiples quorums
+            })
+            .collect();
```

Comment:
This is easy to solve since the nth `OperatorKickParam` is supposed to be for the nth quorum number.

```suggestion
        let operators_to_kick_params: Vec<OperatorKickParam> = operators_to_kick
            .iter()
            .zip(quorum_numbers,iter())
            .map(|(address, quorum_number)| OperatorKickParam {
                operator: *address,
                quorumNumber: *quorum_number,
            })
            .collect();
```